### PR TITLE
Add config-encryption crate for endpoint configs

### PR DESCRIPTION
**Pull request by psFried**
**Description:**

Introduces the flow-config-encryption binary, which is an http server
that encrypts endpoint configurations. The intent is to deploy this as a
cloud-run function. The binary is built into the flow docker image for
expediency, and because it's still unclear what a local setup looks like
with the control plane included. So I decided to keep it easy and close
at hand, until we figure out something better.

The service accepts requests that include both a JSON schema and a
`config` object that conforms to the schema. Assuming the config
validates against the provided schema, it will be encrypted using sops
and returned as either JSON or YAML.

**Workflow steps:**

No user-facing changes.

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/472)

### CODE REVIEW

File: crates/config-encryption/src/encrypt/mod.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        ));
+    }
+    Ok(())
+}
+
+/// We'd like to avoid keeping sensitive values around in memory, but `serde_json::Value` doesn't
+/// have an easy way to zero out the whole document, so this just traverses the object and
+/// overwrites all the String values. There's no safe way to zero numbers, but they seem far less
+/// likely to hold sensitive information anyway.
+fn drop_plain_config(json: serde_json::Value) {
```
Comment (saterus) [inline crates/config-encryption/src/encrypt/mod.rs:194]: Not sure if this helps or not, but you might be able to use https://crates.io/crates/secrecy to help here. It supports deserializing into a `Secret` wrapper that can help with accidentally mishandling a sensitive value. I haven't used it, but it's something you might find interesting.

File: crates/config-encryption/todo.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
+
+- TS Client
+- OpenAPI spec
+- Prometheus metrics
+- CORS headers
+- Tests
+- Build image
+- Deploy
```
Comment (saterus) [inline crates/config-encryption/todo.md:9]: Did you mean to leave this file checked in?

**Review by saterus — APPROVED** [review]
This looks great. I tested it out locally and seems to work great. :tada:

File: crates/config-encryption/src/encrypt/mod.rs (hunk 1 above)
Comment (psFried) [reply crates/config-encryption/src/encrypt/mod.rs:194]: Yeah I had hoped to be able to use something like that, but the reason it doesn't work is that it only handles owned contiguous memory regions. So it won't actually zero the memory of all the nested values when it's dropped :( I could see it still being useful just as a way of communicating to developers that it's sensitive, though. Thanks :+1:

File: crates/config-encryption/todo.md (hunk 2 above)
Comment (psFried) [reply crates/config-encryption/todo.md:9]: lol _totally_ ... I'll remove. Thanks for catching that.

**Pull request merged**